src/Features/LanguageServer/Protocol/Handler/SemanticTokens/CustomLspSemanticTokenNames.cs

```diff
@@ -0,0 +1,150 @@
+        [ClassificationTypeNames.XmlDocCommentAttributeValue] = XmlDocCommentAttributeValue,
+        [ClassificationTypeNames.XmlDocCommentCDataSection] = XmlDocCommentCDataSection,
+        [ClassificationTypeNames.XmlDocCommentComment] = XmlDocCommentComment,
+        [ClassificationTypeNames.XmlDocCommentDelimiter] = XmlDocCommentDelimiter,
+        [ClassificationTypeNames.XmlDocCommentEntityReference] = XmlDocCommentEntityReference,
+        [ClassificationTypeNames.XmlDocCommentName] = XmlDocCommentName,
+        [ClassificationTypeNames.XmlDocCommentProcessingInstruction] = XmlDocCommentProcessingInstruction,
+        [ClassificationTypeNames.XmlDocCommentText] = XmlDocCommentText,
+
+        [ClassificationTypeNames.XmlLiteralAttributeName] = XmlLiteralAttributeName,
+        [ClassificationTypeNames.XmlLiteralAttributeQuotes] = XmlLiteralAttributeQuotes,
+        [ClassificationTypeNames.XmlLiteralAttributeValue] = XmlLiteralAttributeValue,
+        [ClassificationTypeNames.XmlLiteralCDataSection] = XmlLiteralCDataSection,
+        [ClassificationTypeNames.XmlLiteralComment] = XmlLiteralComment,
+        [ClassificationTypeNames.XmlLiteralDelimiter] = XmlLiteralDelimiter,
+        [ClassificationTypeNames.XmlLiteralEmbeddedExpression] = XmlLiteralEmbeddedExpression,
+        [ClassificationTypeNames.XmlLiteralEntityReference] = XmlLiteralEntityReference,
+        [ClassificationTypeNames.XmlLiteralName] = XmlLiteralName,
+        [ClassificationTypeNames.XmlLiteralProcessingInstruction] = XmlLiteralProcessingInstruction,
+        [ClassificationTypeNames.XmlLiteralText] = XmlLiteralText,
+
+        [ClassificationTypeNames.RegexComment] = RegexComment,
+        [ClassificationTypeNames.RegexCharacterClass] = RegexCharacterClass,
+        [ClassificationTypeNames.RegexAnchor] = RegexAnchor,
+        [ClassificationTypeNames.RegexQuantifier] = RegexQuantifier,
+        [ClassificationTypeNames.RegexGrouping] = RegexGrouping,
+        [ClassificationTypeNames.RegexAlternation] = RegexAlternation,
+        [ClassificationTypeNames.RegexText] = RegexText,
+        [ClassificationTypeNames.RegexSelfEscapedCharacter] = RegexSelfEscapedCharacter,
+        [ClassificationTypeNames.RegexOtherEscape] = RegexOtherEscape,
+
+        [ClassificationTypeNames.JsonComment] = JsonComment,
+        [ClassificationTypeNames.JsonNumber] = JsonNumber,
+        [ClassificationTypeNames.JsonString] = JsonString,
+        [ClassificationTypeNames.JsonKeyword] = JsonKeyword,
+        [ClassificationTypeNames.JsonText] = JsonText,
+        [ClassificationTypeNames.JsonOperator] = JsonOperator,
+        [ClassificationTypeNames.JsonPunctuation] = JsonPunctuation,
+        [ClassificationTypeNames.JsonArray] = JsonArray,
+        [ClassificationTypeNames.JsonObject] = JsonObject,
+        [ClassificationTypeNames.JsonPropertyName] = JsonPropertyName,
+        [ClassificationTypeNames.JsonConstructorName] = JsonConstructorName,
+    }.ToImmutableDictionary();
```

Comment:
if generators were easier, this would be totally fully doable with partial props :)

Reply:
consider a test, or debug-only static-constructor checks that all fields have values that match what is expected, and taht this dictionary contains all fields.

Reply:
The construction of the schema will fail if not all classification type names are mapped, which will cause all the LSP tests to fail.  So I think we're good there.